dvc/repo/imp_url.py

```diff
@@ -34,6 +36,35 @@ def imp_url(
     ):
         url = relpath(url, wdir)
 
+    if glob:
+        from glob import glob
+
+        abs_url = os.path.join(erepo["url"], url)
+        expanded_targets = [
+            entry.replace(erepo["url"] + os.path.sep, "")
+            for entry in glob(abs_url, recursive=True)
+        ]
+        expanded_targets = [
+            t for t in expanded_targets if not t.endswith(".dvc")
+        ]
+
+        if len(expanded_targets) != 1:
+            msg = (
+                "Cannot import multiple files or directories at once"
+                "See {} for more information."
+            ).format(
+                format_link(
+                    "https://dvc.org/doc/user-guide/"
+                    "troubleshooting#import-wildcard"
+                )
+            )
+            raise DvcException(msg)
```

Comment:
Hmmm so what's the point of wildcards in `import`? 🙂

Reply:
p.s. will this affect `import-url`? How about `get` and `get-url`? Thanks. Cc @efiop

Reply:
Yeah, in case of `import` it is only useful for local git repos, so more of a synthetic use case. Also the current implementation doesn't work for `import` at all, because for `import` there are no local files that iglob could work on, because the target repo is not even cloned yet. Not sure how the tests work here.

Reply:
> p.s. will this affect import-url? How about get and get-url? Thanks. Cc @efiop

Yes it will. No, it won't affect get and get-url, at least right now. In the future get and get-url should support glob too.

Reply:
> Seems like the current implementation doesn't work for `import` at all, because for `import` there are no local files that iglob could work on, because the target repo is not even cloned yet. Not sure how the tests work here.

I got a bit confused myself when I implemented this, and I digged deeper into how import works, and for this test `test_import_wildcard`, here is my understanding:

- there is a repository `erepo_dir` created locally with 3 files: `foo`, `foo123` and `efoothisistoocomplicatedtotype`; this is created somewhere locally, e.g.: `'/private/var/folders/_3/lmzh2stn56z_459q83p871_h0000gn/T/pytest-of-juogri/pytest-122/erepo0'`
- the command issued for importing `efoo*` as `efoo_imported` 
- the current implementation uses absolute path for the repository to do globbing; feels hacky and only works if the repo exists locally
- resolves the path to `'/private/var/folders/_3/lmzh2stn56z_459q83p871_h0000gn/T/pytest-of-juogri/pytest-122/erepo0/efoo*'`, and then finds `'efoothisistoocomplicatedtotype'`

What would be a less hacky approach here that would work for all import cases ?

Reply:
It needs to do virtual glob on the cloned repo, which means that this PR will need to be re-done from scratch. Let's close this PR for now, since the most useful part is not really implemented.